### PR TITLE
Fix: handle NULL offset of NTH_VALUE window function

### LINKED ISSUE #12815: Panic in `nth_value` window function (SQLancer)
### Describe the bug

See the reproducer in datafusion-cli (compiled from the latest main, commit https://github.com/apache/datafusion/commit/84c94097896e9d08e40d63e8c505414ea70e8b29)

```shell
/*DML*/CREATE TABLE t41(v0 DOUBLE, v1 BOOLEAN, v2 STRING, v3 STRING);
/*DML*/INSERT INTO t41(v0, v2) VALUES ('NaN'::Double, 'd殪s7'), (0.4097890978398071, '顼)j');

> SELECT NTH_VALUE(tt0.v0, NULL) OVER (PARTITION BY tt0.v2 ORDER BY tt0.v0) FROM t41 AS tt0;
thread 'main' panicked at /Users/yongting/Desktop/code/my_datafusion/arrow-datafusion/datafusion/physical-expr/src/window/nth_value.rs:195:63:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

### To Reproduce

_No response_

### Expected behavior

Returns 2 rows of `Null`
Run in duckdb:
```shell
D /*DML*/CREATE TABLE t41(v0 DOUBLE, v1 BOOLEAN, v2 STRING, v3 STRING);
D /*DML*/INSERT INTO t41(v0, v2) VALUES ('NaN'::Double, 'd殪s7'), (0.4097890978398071, '顼)j');
D SELECT NTH_VALUE(tt0.v0, NULL) OVER (PARTITION BY tt0.v2 ORDER BY tt0.v0) FROM t41 AS tt0;
┌────────────────────────────────────────────────────────────────────┐
│ nth_value(tt0.v0, NULL) OVER (PARTITION BY tt0.v2 ORDER BY tt0.v0) │
│                               double                               │
├────────────────────────────────────────────────────────────────────┤
│                                                                    │
│                                                                    │
└────────────────────────────────────────────────────────────────────┘
```

### Additional context

Found by SQLancer https://github.com/apache/datafusion/issues/11030

### PR DESCRIPTION
**Pull request by HuSen8891**
## Which issue does this PR close?

Closes #12815

### CODE REVIEW

**Comment by jonathanc-n** [comment]
@HuSen8891 This lgtm!

File: datafusion/sqllogictest/test_files/window.slt [hunk 1]
```diff
@@ -4929,6 +4929,15 @@ SELECT v1, NTH_VALUE(v2, 0) OVER (PARTITION BY v1 ORDER BY v2) FROM t;
 2 NULL
 2 NULL
 
+query I
+SELECT NTH_VALUE(tt0.v1, NULL) OVER (PARTITION BY tt0.v2 ORDER BY tt0.v1) FROM t AS tt0;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/window.slt:4933]: I double checked that without the change in this PR, this test panics as described in the ticket:

```
thread 'tokio-runtime-worker' panicked at datafusion/physical-expr/src/window/nth_value.rs:195:63:
attempt to subtract with overflow
stack backtrace:
   0: rust_begin_unwind
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
```

**Review by alamb — APPROVED** [review]
Thank you @HuSen8891  and @jonathanc-n  -- this looks great 🦾

**Comment by alamb** [comment]
Thanks agin @jonathanc-n

**Pull request merged**